### PR TITLE
Revise SDL-0296 - Possibility to update video streaming capabilities during ignition cycle

### PR DESCRIPTION
**Pull request by dboltovskyi**
The main purpose of this proposal is to make it possible to update video streaming capabilities during the ignition cycle.

This is a new revision for initial version [0296-Update-video-streaming-capabilities-during-ignition-cycle](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md) which is mostly address the comments provided in https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-645074226.

It includes the following changes:
 - Re-worked `Motivation` section with examples of use cases
 - Split `Proposed solution` section into 2 parts:
   - Negotiation of VSCs - describes the changes in APIs related to available (HMI) and supported (App) VSCs
   - Resolution Switching - describes the flow when HMI initiate changes in VSCs
- Provided examples of the changes in SDL Mobile Libraries and Public APIs
- Added section for Old applications behavior
- Added section for HMI Guidelines
- Updated `Alternatives considered` section

### CODE REVIEW

**Comment by dboltovskyi** [comment]
@theresalech Please notice here is new version of proposal that we would like to bring up for review.

**Comment by dboltovskyi** [comment]
@theresalech Please notice new revision is ready for PM review.

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 1]
```diff
@@ -2,76 +2,216 @@
...
+ - Collapsed view
+
+The idea of the proposal is to add such possibility into SDL ecosystem. Also offered solution would allow to adjust size of UI elements depends on a current screen mode.
 
 ## Proposed solution
 
 ### SDL Core changes
 
-In order to make it possible to update the described parameters during a single ignition cycle, existing `OnSystemCapabilityUpdated` notification can be used.
-Currently, `REMOTE_CONTROL` and `APP_SERVICES` capabilities can already be updated through it, so the idea is to extend this list by `VIDEO_STREAMING` capabilities.
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:24]: Adding `DISPLAYS` into processing by `OnSystemCapabilityUpdated` notification is not necessary anymore.

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 2]
```diff
@@ -2,76 +2,216 @@
...
+
+<struct name="AppCapability" since="x.x">
+    <param name="appCapabilityType" type="AppCapabilityType" mandatory="true">
+        <description>Used as a descriptor of what data to expect in this struct. The corresponding param to this enum should be included and the only other param included.</description>
+    </param>
+    <param name="videoStreamingCapability" type="VideoStreamingCapability" mandatory="false">
+        <description>Describes supported capabilities for video streaming </description>
+    </param>
+</struct>
+
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:66]: Yes, that's right. `AppVideoStreamingCapability` struct is not used in solution we agreed. Please see item 6 of https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-639089698. It seems this struct was copied from one of the previous intermediate solutions.

**Comment by dboltovskyi** [comment]
@theresalech Thank you a lot for all your suggestions. They were applied in b6fcf68.
Please also find responses for the two questions.

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 3]
```diff
@@ -11,28 +11,28 @@ The main purpose of this proposal is to make it possible to update video streami
 
 ## Motivation
 
-Currently there are numerous VSC parameters may be provided by HMI through SDL to Mobile application.
-These may also include the ones described in [Pixel density and Scale](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0179-pixel-density-and-scale.md) proposal.
-These parameters may be used by Projection or Navigation mobile application to properly adjust video streaming data.
-Unfortunately, once these parameters defined during HMI initialization process, they can't be updated until the next ignition cycle.
+Currently there are numerous VSC parameters which may be provided by the HMI through SDL Core to Mobile applications.
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:14]:
```suggestion
Currently there are numerous VSC parameters which may be provided by the HMI through SDL Core to mobile applications.
```

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 4]
```diff
@@ -76,36 +76,36 @@ Within this request mobile application (SDL library) can also subscribe to recei
 </struct>
 ```
 
-Details on how mobile application selects supported VSCs described in [Mobile libraries changes](0296-Update-video-streaming-capabilities-during-ignition-cycle.md#mobile-libraries-changes-ios-and-android) section.
+Details on how a mobile application selects supported VSCs is described in the [Mobile libraries changes](0296-Update-video-streaming-capabilities-during-ignition-cycle.md#mobile-libraries-changes-ios-and-android) section.
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:79]:
```suggestion
Details on how a mobile application selects supported VSCs are described in the [Mobile libraries changes](0296-Update-video-streaming-capabilities-during-ignition-cycle.md#mobile-libraries-changes-ios-and-android) section.
```

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (theresalech) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:66]: While it's possible there was a discrepancy in item 6 of the [requested revisions](https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-642762858), these revisions are what the SDLC voted on and therefore must be included in this PR.  If you'd like to suggest removing the struct, that's okay, but would need to be done in a new comment on the proposal review issue once the revised proposal is back in review.

[the remark this answers is not in the document]
Comment (dboltovskyi) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:66]: Ok. Let's do in this way. Please find update in 0fc100b

**Comment by dboltovskyi** [comment]
@theresalech Thank you for your review! Please notice all comments are addressed.

**Pull request merged**